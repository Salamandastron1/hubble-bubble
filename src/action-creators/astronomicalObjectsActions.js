export const starsReceived = astronomicalObjects => ({
  type: 'STARS_RECEIVED',
  astronomicalObjects,
})

export const toggleSelected = id => ({
  type: 'TOGGLE_SELECTED',
  id,
})