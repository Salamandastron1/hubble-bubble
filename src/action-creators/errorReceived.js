export const errorReceived = error => ({
  type: 'ERROR_RECEIVED',
  error
})