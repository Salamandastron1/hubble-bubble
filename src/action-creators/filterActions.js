export const showAll = () => ({
  type: 'SHOW_ALL'
})

export const showGame = () => ({
  type: 'SHOW_GAME',
})

export const showMatches = () => ({
  type: 'SHOW_MATCHES'
})