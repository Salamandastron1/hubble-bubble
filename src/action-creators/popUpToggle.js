export const popUpToggle = () => ({
  type: 'POPUP_TOGGLE',
})