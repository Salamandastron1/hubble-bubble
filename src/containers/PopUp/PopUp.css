.popup-outer {
  position: fixed;
  height: 100%;
  width: 100%;
  background-color: rgba(0,0,0, 0.5);
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  font-family: 'Roboto'; 
}

.popup-inner {
  position: absolute;
  left: 15%;
  right: 15%;
  top: 15%;
  bottom: 15%;
  background: #e5c5c4;
  overflow-y: scroll;
  display: grid;
  justify-items: center;
  padding: 25px;
  border-radius: 25px;
  grid-template-areas:
  'a a a' 
  'e e e'
  'b b b'
  'c c c';
}
.response {
  grid-area: a;
  color: #ba57a6;
  font-family: 'Titan One';
  font-size: 2em;
}
.pop-up-image {
  background-size: cover;
  border-radius: 10px;
  height: 500px;
  width: 500px;
  grid-area: b;
}
.description {
  grid-area: c;
  color: #ba57a6;
  font-family: 'Roboto';
  font-size: 1.5em;
  padding: 50px; 
}
.next-question {
  position: absolute;
  left: 25px;
  top: 25px;
  background: #ba57a6;
  border: none;
  color: white;
  height: 35px;
  width: 35px;
  font-size: 1.5em;
  border-radius: 50%;
}
.name-star {
  grid-area: e;
  text-align: center;
  color: #ba57a6;
  font-size: 2em;
  border-radius: 15px;
}

@media (min-width: 320px) and (max-width: 540px) {
  .popup-inner {
    display: flex;
    flex-direction: column;
    left: 5%;
    right: 5%;
    top: 5%;
    bottom: 5%;
  }
  popup-image{
    height: 50px;
    width: 50px;
  }
}


