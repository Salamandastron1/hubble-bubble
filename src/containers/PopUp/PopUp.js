import React from 'react'
import { popUpToggle } from '../../action-creators/popUpToggle'
import { connect } from 'react-redux'
import './PopUp.css'

export const PopUp = (props) => {
  const { star, nextQuestion, closePopUp, correct } = props
  return (
    <section className='popup-outer'>
      <div className='popup-inner'>
      {correct ?
        <h1>WOW YOU'RE RIGHT</h1>
        : <h1>Noooooot quite :)</h1>}
        <h2>{star.name}</h2>
        <button
          onClick={() => {
            closePopUp();
            nextQuestion()
          }}
        >
          Next Question
        </button>
        <img 
          src={star.image_files} 
          height='200'
          width='300' />
        <p>{star.description}</p>
      </div>
    </section>
  )
}

export const mapDispatchToProps = dispatch => ({
  closePopUp: () => dispatch(popUpToggle())
})

export default connect(undefined, mapDispatchToProps)(PopUp)