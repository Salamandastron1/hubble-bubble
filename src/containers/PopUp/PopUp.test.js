import { PopUp, mapDispatchToProps } from './PopUp'
import React from 'react'
import {shallow} from 'enzyme'
import { popUpToggle } from '../../action-creators/popUpToggle'

it('should match the snapshot', () => {
  const mockRefresh = jest.fn()
  const wrapper = shallow(<PopUp 
    star={{name: 'name', description: 'blah', image_files: 'url', id: 4}}
    refreshState={mockRefresh}/>)

  expect(wrapper).toMatchSnapshot()
})
it('should match the snapshot if correct is truthy', () => {
  const wrapper = shallow(<PopUp 
    star={{name: 'name', description: 'blah', image_files: 'url', id: 4}}
    correct={true}/>)

  expect(wrapper).toMatchSnapshot()
})
it('should call mockRefresh on button click', () => {
  const mockRefresh = jest.fn()
  const mockPop = jest.fn()
  const wrapper = shallow(<PopUp 
    star={{name: 'name', description: 'blah', image_files: 'url', id: 4}}
    nextQuestion={mockRefresh}
    closePopUp={mockPop}/>)
  wrapper.find('button').simulate('click')

  expect(mockRefresh).toHaveBeenCalled()
})
it('should call dispatch on button click', () => {
  const mockDispatch = jest.fn()
  const mockRefresh = jest.fn()
  const wrapper = shallow(<PopUp 
    star={{name: 'name', description: 'blah', image_files: 'url', id: 4}}
    closePopUp={mockDispatch}
    nextQuestion={mockRefresh}/>)
  wrapper.find('button').simulate('click')

  expect(mockDispatch).toHaveBeenCalledWith()
})

it('should call dispatch with the correct params', () => {
  const mockDispatch = jest.fn()
  const mappedProps = mapDispatchToProps(mockDispatch)

  mappedProps.closePopUp()

  expect(mockDispatch).toHaveBeenCalledWith(popUpToggle())
})



