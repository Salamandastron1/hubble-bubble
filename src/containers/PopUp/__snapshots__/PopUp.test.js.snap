// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section
  className="popup-outer"
>
  <div
    className="popup-inner"
  >
    <h2
      className="response"
    >
      Noooooot quite :)
    </h2>
    <h2
      className="name-star"
    >
      name
    </h2>
    <button
      className="next-question"
      onClick={[Function]}
    >
      X
    </button>
    <div
      className="pop-up-image"
      style={
        Object {
          "backgroundImage": "url(url)",
        }
      }
    />
    <p
      className="description"
    >
      blah
    </p>
  </div>
</section>
`;

exports[`should match the snapshot if correct is truthy 1`] = `
<section
  className="popup-outer"
>
  <div
    className="popup-inner"
  >
    <h2
      className="response"
    >
      WOW YOU'RE RIGHT
    </h2>
    <h2
      className="name-star"
    >
      name
    </h2>
    <button
      className="next-question"
      onClick={[Function]}
    >
      X
    </button>
    <div
      className="pop-up-image"
      style={
        Object {
          "backgroundImage": "url(url)",
        }
      }
    />
    <p
      className="description"
    >
      blah
    </p>
  </div>
</section>
`;
